### PR TITLE
Product filters 

### PR DESCRIPTION
**Pull request by saicharan-gouru**
- Refactored The code and optimized the code
- Added sort by price functionality

Spazeir-mart-productfilter

### CODE REVIEW

File: src/contexts/product-context.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+function DataProvider({children}){
+
+    const [{products,categories,sortByPrice},productsDispatch] = useReducer(productsReducer,{ products:[],categories:[] ,sortByPrice : null });
+
+    useEffect(()=>{ 
+        async function FetchData(){
+            try{
+                const response1 = await axios.get("/api/categories");
+                productsDispatch({type:"FETCH_CATEGORIES",payload:response1.data.categories})
+                const response2 = await axios.get("/api/products");
```
Comment (Aditi0505) [inline src/contexts/product-context.js:18]: Can have a descriptive name here in place of response1 and response2

File: src/pages/productlistingpage/components/Filters.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        <div className="filters">
+            <h4>Filters <a href="/" className="filter-clear">clear</a></h4>
+            <h5>Sort by</h5>
+            <label> <input type="radio" name="sort-by" checked={sortByPrice === "LOW_TO_HIGH"} onChange={()=>productsDispatch({type:"SORT_BY_PRICE",payload:"LOW_TO_HIGH"})} /> price - low to high</label><br/>
+            <label> <input type="radio" name="sort-by" checked={sortByPrice === "HIGH_TO_LOW"} onChange={()=>productsDispatch({type:"SORT_BY_PRICE",payload:"HIGH_TO_LOW"})}/> price - high to low</label><br/>
+            <h5>Category</h5>
+            <label><input type="checkbox" name="staples" /> Staples</label><br/>
+            <label><input type="checkbox" name="snacks" /> Snacks</label><br/>
+            <label><input type="checkbox" name="diary-and-eggs" /> Diary and eggs</label><br/>
+            <label><input type="checkbox" name="vegetables" /> Vegetables</label><br/>
```
Comment (Aditi0505) [inline src/pages/productlistingpage/components/Filters.js]: You can map over a list here to avoid code repetition.

**Review by Aditi0505 — APPROVED** [review]
Left few suggestions.
Another suggestion would be to use a code formatter. 

Things that I liked:
1. Good folder structure
2. Good use of context and reducers

**Review by anupkgurung — APPROVED** [review]
good structure.
clean code

**Pull request merged**